scripts/imgtool/design.md

```diff
@@ -244,6 +245,30 @@ work properly even when it is reset during the middle of an image swap. For this
 reason, the rest of the document describes its behavior when configured to swap
 images during an upgrade.
 
+### [RAM Loading](#ram-loading)
+
+In ram-loading mode the slots are equal. Like the direct-xip mode, this mode
+also selects the newest image by reading the image version numbers in the image
+headers. But instead of executing it in place, the newest image is copied to the
+RAM for execution. The load address, the location in RAM where the image is
+copied to, is stored in the image header. The ram-loading upgrade mode can be
+useful when there is no internal flash in the SoC, but there is a big enough
+internal RAM to hold the images. Usually in this case the images are stored
+in an external storage device. Execution from external storage has some
+drawbacks (lower execution speed, image is exposed to attacks) therefore the
+image is always copied to the internal RAM before the authentication and
+execution. Ram-loading mode requires the image must be built to be executed
+from the RAM address range instead of the storage device address range. If
+ram-loading is enabled then platform must define the following parameters:
+
+```c
+#define IMAGE_EXECUTABLE_RAM_START    <area_base_addr>
+#define IMAGE_EXECUTABLE_RAM_SIZE     <area_size_in_bytes>
+```
+
```

Comment:
It would be helpful to mention that one must use the `--load-addr` option of imgtool when signing the images for the ram-loading strategy. 
```
When ram-loading is enabled, the `--load-addr <addr>` option of the imgtool script must also be used when signing the images. This option will add the RAM_LOAD flag to the image header which indicated that this image should be loaded into RAM and also adds the load address to it.
```

Reply:
ack